### ISSUE TITLE
Header background coming down with scrolling

### ISSUE THREAD

**Issue by knod** [omriAckley/qromp#14]
The header background seems to be fixed. This may be because of absolute positioning or some such, and #scroller might have been controlling that, but the header text and the header shouldn't be separate anyway. The site generally needs to be rebuilt, so that can be taken care of then.

**Comment by sethtoles** [comment]
Yeah, that's exactly what `#scroller` was doing. The issue arose because all of the page content _except_ `#header-back` has a maximum width and needs to be centered, so I gave it `margin: auto`. `#header-back` is the only element which needs to be the full page width, so it's in a separate div with `width: 100%`. To get `#header-back` to sit behind the app, you could either give it `position: absolute`, or give the rest of the app `margin-top: -4.6rem`, 4.6 rems being the height of `#header-back`.

However, if you resize the window to be narrower than the app, all of a sudden, you get horizontal scrolling. You need this to be able to see all of the app, but when you scroll right, you see the edge of `#header-back`, because it's only 100% the width of the window, and now you've scrolled past that point. You could make header-back `position: fixed`, so it sticks to your view when you scroll sideways, but then it also sticks to you view when you scroll down.

My solution: `#header-back` can sit at the top of the page, behind the rest of the content, taking up the full width with `position:absolute; width:100%`. Then all you need to do is make sure it doesn't scroll sideways with the rest of the app, so you put the rest of the app in `#scroller`, which has `overflow-x: auto`. When the page gets too narrow, ONLY the stuff in `#scroller` can scroll sideways, so `#header-back` is never moved off the page.

This is what I could come up with. There may very well be another, better way.
Sorry for the long text, I didn't want to just say, "We need it," and leave you trying to figure out why.

**Comment by sethtoles** [comment]
Also, I'm curious what you think needs to be rebuilt about the rest of the HTML. There are two eccentricities I can think of, but they are also born from weird requirements. Maybe I'm missing something. I _do_ think that a lot of the content should be put in dynamically, but I've already started to work on that.

**Comment by knod** [comment]
Thank you for all that explanation. The header should probably be done as a regular header. Now that we're using sass we can just make the margins the same for the header stuff and the rest of the body.

As for what else needs to be changed - a lot of it is stuff I just learned, so I'm sorry if I came off as flippant. Also, some of it was feedback from people on the irc channel, but when I looked over it more carefully I thought it looked like they just didn't understand the functionality of the website.

Dynamic content sounds good, though I think a lot of the stuff that's there makes sense. I haven't looked much into best practices on dynamic content.

We can talk more about what I've learned, though I think other than the header things are fine for now since there isn't much time. The header is currently an issue because whenever I put any content in the "about" div, the vertical scrolling problem shows up.
